pkg/kube/client_factory.go

```diff
@@ -85,20 +84,8 @@ func (c *clientFactory) ToRESTMapper() (meta.RESTMapper, error) {
 	if err != nil {
 		return nil, err
 	}
-	rc, err := c.ToRESTConfig()
-	if err != nil {
-		return nil, err
-	}
 	c.mapperOnce.Do(func() {
-		c.mapper, _ = apiutil.NewDynamicRESTMapper(rc, apiutil.WithLazyDiscovery, apiutil.WithCustomMapper(func() (meta.RESTMapper, error) {
-			discoveryClient.Invalidate()
-			// Use a custom mapper so we can re-use our discoveryClient instead of creating a new one
-			groupResources, err := restmapper.GetAPIGroupResources(discoveryClient)
-			if err != nil {
-				return nil, err
-			}
-			return restmapper.NewDiscoveryRESTMapper(groupResources), nil
-		}))
+		c.mapper = restmapper.NewDeferredDiscoveryRESTMapper(discoveryClient)
```

Comment:
It seems that `discoveryClient.Invalidate()` has been removed, should it be added before `discoveryClient` is passed to `NewDeferredDiscoveryRESTMapper`? or some else where this client is really used?

Reply:
it's moved to applyYaml

Reply:
oh, sorry for missing that part.